test/automate/automate-utils.js

```diff
@@ -14,285 +14,292 @@
     .build();
 
   // The following functions are defined below instead of outside this function
   // so we do not have to pass the 'driver' as a function parameter. It makes
   // the functions easier to use in a 'then' chain.
 
+  // Grab a screenshot and write to disk.
+  // TODO: Do we want to grab screenshots for success?
+  const saveScreenshot = (name, type) => {
+
+    const fileName =
+      `test-result-${screenShotUniq}-${browser}-` +
+      `${name.replace(/[ :]/g, '_')}-${type}.png`;
+
+    screenShotUniq++;
+
+    return driver.takeScreenshot()
+      .then(data => {
+        if (data) {
+          var base64Data = data.replace(/^data:image\/png;base64,/, '');
+          return fs.writeFile(
+            path.join(screenshotDir, fileName),
+            base64Data,
+            'base64',
+            err => {if (err) { console.log(err); }}
+          );
+        }
+        return null;
+      });
+  };
+
   // Locates an element and then clicks it.
-  const clickElement = function(locator) {
-    return driver.wait(
+  const clickElement = locator => driver.wait(
       until.elementLocated(locator),
       elementTimeout
     )
-    .then(clickWhenVisible, errorFunc);
-  };
+    .then(clickWhenVisible);
 
   // Waits for an element to be visible, and then clicks it.
-  const clickWhenVisible = function(element) {
-    return driver.wait(
-     until.elementIsVisible(element),
-     elementTimeout
-   )
-   .then(function(element) {
-     element.click();
-     return element;
-   },
-   errorFunc);
-  };
+  const clickWhenVisible = element => driver.wait(
+      until.elementIsVisible(element),
+      elementTimeout
+    )
+    .then(element => element.click().then(() => element));
 
   // Waits for an element to have opacity of 1. Helps for waiting for elements
   // that fade in. Returns the element when the opacity reaches '1' so that
   // this can be chained with other promises.
-  const waitOpacity = function(element) {
-    return driver.wait(function() {
-      return element.getCssValue('opacity')
-      .then(function(opacity) {
-        if (opacity === '1') {
-          return element;
-        } else {
-          return false;
-        }
-      });
-    });
-  };
+  const waitOpacity = element => driver.wait(() => (
+    element.getCssValue('opacity')
+    .then(opacity => (opacity === '1') ? element : false)
+  ));
 
   // Handles errors. Saves a screenshot so we can see what the error is.
-  const errorFunc = function(error) {
-    // Ignore unexpected alert error, log all others.
-    if (error.name === 'UnexpectedAlertOpenError') {
-      console.log('Ignoring UnexpectedAlertOpenError');
-      saveScreenshot(driver, screenshotDir, name, `ERROR-${error.name}`);
-    } else {
-      saveScreenshot(driver, screenshotDir, name, `ERROR-${error.name}`);
+  const errorFunc = error => {
       console.error(error);
-    }
+      return saveScreenshot(name, `ERROR-${error.name}`);
+  };
+
+  // Alert appears only once.
```

Comment:
This looks like another improvement? I think we were having problems in the past with the tests failing due to alerts? Good improvement, but like above, please try to put improvements in a separate PR in the future.

Reply:
Before deployment of this PR, tests don't pass without this.